api/v1alpha1/validation.go

```diff
@@ -116,3 +135,52 @@ func (r ResourceSync) Validate() []error {
 	allErrs = append(allErrs, validation.ValidateString(&r.Spec.Path, "spec.path", 0, 2048, nil, "")...)
 	return allErrs
 }
+
+func validateGitHttpConfig(config *GitHttpConfig) []error {
+	var errs []error
+	if config != nil {
+		if config.CaCrt != nil {
+			if !validation.IsBase64(*config.CaCrt) {
+				errs = append(errs, fmt.Errorf("httpConfig.caCrt must be a valid base64 encoded string"))
+			}
+		}
+
+		if config.TlsCrt != nil {
+			if !validation.IsBase64(*config.TlsCrt) {
+				errs = append(errs, fmt.Errorf("httpConfig.tlsCrt must be a valid base64 encoded string"))
+			}
+		}
+
+		if config.TlsKey != nil {
+			if !validation.IsBase64(*config.TlsKey) {
+				errs = append(errs, fmt.Errorf("httpConfig.tlsKey must be a valid base64 encoded string"))
+			}
+		}
+
+		if config.Username != nil {
+			errs = append(errs, validation.ValidateString(config.Username, "httpConfig.username", 1, 256, nil, "")...)
+		}
+
+		if config.Password != nil {
+			errs = append(errs, validation.ValidateString(config.Password, "httpConfig.password", 1, 256, nil, "")...)
+		}
+	}
```

Comment:
Would it make sense to also test for which fields go/don't go together? E.g. user+pass must be given both or nothing, either TLS or user+pass must be provided but not both?

Reply:
I think valid options are:

for http config:
 - user + pass
 - tlsCrt + tlsKey
 - user + pass + tlsCrt + tlsKey (yes, you can have combination)
 - other fields are optional

for sshConfig:
 - passphrase cannot be specified without private key